packages/api/src/graphql/comment/comment.private-mutation.ts

```diff
@@ -19,3 +25,85 @@ export const takeActionOnComment = (
     }
   })
 }
+
+export const updateComment = async (
+  commentId: string,
+  text: string | undefined,
+  tagIds: string[] | undefined,
+  authenticate: Context['authenticate'],
+  commentClient: PrismaClient['comment']
+) => {
+  const {roles} = authenticate()
+  authorise(CanUpdateComments, roles)
+
+  return commentClient.update({
+    where: {id: commentId},
+    data: {
+      revisions: text
+        ? {
+            create: {
+              text
+            }
+          }
+        : undefined,
+      tags: tagIds
+        ? {
```

Comment:
quite unreadable. I would refactor that in a separate variable before commentClient.update() method.